emulode/ode.py

```diff
@@ -75,6 +75,107 @@ def rossler(t: float, y: np.ndarray, params: dict[str, float]) -> np.ndarray:
             ]
         )
 
+    @staticmethod
+    def seir_freq(t: float, y: np.ndarray, params: dict[str, float]) -> np.ndarray:
+        """
+        SEIR system with frequency depdendent force of infection,
+        demography, disease induced death and loss of immunity.
+        """
+        # raise NotImplementedError("This is a work in progress")
+        # print("I will come back to this code later")
+
+        Utils.check_parameters(
+            params, ["PI", "mu", "beta", "sigma", "gamma", "epsilon", "alpha"]
+        )
+        Utils.check_dimension(y, 4)
+        N = y[0] + y[1] + y[2] + y[3]  # total population
+        return np.array(
+            [
+                params["PI"]
+                - params["beta"] * y[1] * y[0] / N
+                - params["mu"] * y[0]
+                + params["alpha"] * y[3],
+                params["beta"] * y[1] * y[0] / N
+                - -params["sigma"] * y[1]
+                - params["mu"] * y[1],
+                params["sigma"] * y[1] - params["gamma"] * y[2] - params["mu"] * y[2],
+                params["gamma"] * (1 - params["epsilon"]) * y[2]
+                - params["mu"] * y[3]
+                - params["alpha"] * y[3],
+            ]
+        )
+
+    @staticmethod
+    def seir_dens(t: float, y: np.ndarray, params: dict[str, float]) -> np.ndarray:
+        """SEIR system with density depdendent force of infection, demography, disease induced death and loss of immunity."""
```

Comment:
It's a silly 'rule' but long lines are to be avoided. Basically because it makes it difficult to read when multiple windows are open side-by-side. Make it into a multi-line docstring. See example in lines 80-83
```suggestion
        """
        SEIR system with density depdendent force of infection, 
        demography, disease induced death and loss of immunity.
        """
```